pkg/webhook/cluster/validation/validation.go

```diff
@@ -121,3 +134,53 @@ func (v *validator) buildValidationDependencies(ctx context.Context, c *kubermat
 
 	return datacenter, cloudProvider, nil
 }
+
+func (v *validator) validateProjectRelation(ctx context.Context, cluster *kubermaticv1.Cluster, oldCluster *kubermaticv1.Cluster) *field.Error {
+	label := kubermaticv1.ProjectIDLabelKey
+	fieldPath := field.NewPath("metadata", "labels")
+
+	if oldCluster != nil && cluster.Labels[label] != oldCluster.Labels[label] {
+		return field.Invalid(fieldPath, cluster.Labels[label], fmt.Sprintf("the %s label is immutable", label))
+	}
+
+	// during cluster creation, we enforce the project label;
+	// during updates we are more relaxed and only require that the label isn't changed,
+	// so that if a project gets removed before the cluster (for whatever reason), then
+	// the cluster cleanup can still progress and is not blocked by the webhook rejecting
+	// the stale label
+	if oldCluster == nil {
```

Comment:
what do you think about skipping the update check only if the project doesnt exist?